source/computer/macos-setup.rst

```diff
@@ -1,9 +1,171 @@
 macOS 配置指南
 ==============
 
+:本节作者: `王亮 <https://github.com/wangliang1989>`__\（作者）、
+           `田冬冬 <https://me.seisman.info/>`__\（作者）、
+           `姚家园 <https://github.com/core-man>`__\ （审稿）
+:最后更新日期: 2021-01-11
+
+----
+
+.. warning::
+
+   本文成稿于 2018 年 1 月，并于 2021 年 1 月做少量改动。
+   但笔者尚未在新电脑上进行测试，无法保证所有命令均正确。欢迎反馈。
+
+第一次启动 Mac 电脑后，经过简单的设置，就得到了一个可供日常使用的 macOS 系统，
+但这样的 macOS 系统尚不能满足日常科研与编程开发的需求，还需要做进一步的配置。
+
+Command Line Tools for Xcode
+----------------------------
+
+Xcode 是 macOS 下的集成开发环境（IDE），类似于 Windows 下的 Microsoft Visual Studio。
+Command Line Tools for Xcode 是 Xcode 的一个子集，其包含了常用的命令行开发工具，
+比如 C 编译器、make、git 等，是安装很多其他软件的基础。
+
+按下 :kbd:`Command+空格`\ ，搜索并打开“Terminal”。在 Terminal 中执行如下命令
+以安装 Command Line Tools for Xcode::
+
+   $ xcode-select --install
+
+在弹出的窗口点击“Install”即可。
+
+此处安装的 Command Line Tools 可能不是最新版，可以到“App Store”的“更新”中查看
+是否有相关更新。如果有，则升级到最新版。
+
+Homebrew
+--------
+
+`Homebrew <https://brew.sh/index_zh-cn.html>`__ 是 macOS 下的第三方软件包管理器。
+可以用于安装各种常见的软件包、库文件以及字体。日常及科研工作中所需的大多数软件、
+库文件以及字体都可以通过 Homebrew 安装。
+
+要安装 Homebrew，需要将如下命令复制到终端中并执行::
+
+    /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+
+Homebrew 的详细用法见官方文档。此处仅列出一些常用的用法::
+
+    # 模糊搜索与 java 相关的软件
+    $ brew search java
+
+    # 安装 java 软件包
+    $ brew install java
+
+    # 安装 QQ
+    # 由于 QQ 是带图形界面的软件，因而这里需要使用 --cask 选项
+    $ brew install --cask qq
+
+    # 升级某个软件
+    $ brew upgrade xxx
+
+    # 卸载某个软件
+    $ brew uninstall xxx
+
 .. note::
 
-   尚未完成。
+   Homebrew 可能在国内下载速度较慢，可以考虑使用中科大 LUG 提供的 Homebrew 镜像。
+   详细使用方法见：
+
+   - Homebrew 源镜像：http://mirrors.ustc.edu.cn/help/brew.git.html
+   - Homebrew Bottles 源镜像：http://mirrors.ustc.edu.cn/help/homebrew-bottles.html
+   - Homebrew Core 源镜像：http://mirrors.ustc.edu.cn/help/homebrew-core.git.html
+   - Homebrew Cask 源镜像：http://mirrors.ustc.edu.cn/help/homebrew-cask.git.html
```

Comment:
Maybe also explain what "formulae", "bottles", "core" and "cask" mean in Homebrew.

Reply:
Yes.

Reply:
I just don't know what are bottle core and xask

Reply:
Will give a brief explanation later.